staging/src/k8s.io/apiserver/pkg/endpoints/handlers/workflow.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"net/http"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// guessWorkflowID tries to find a reasonable default for the
```

Comment:
Yeah, I selected the word "guess" rather than "default"  since it's not a hard-coded value, but something that has a poor heuristic. I still think "guess" is a better description of what this does ;-)